gammapy/irf/tests/test_background.py

```diff
@@ -49,3 +53,175 @@ def test_background_3d_write(bkg_3d):
+    res = bkg_3d.evaluate(fov_offset=offset_tab, fov_phi=phi_tab, energy_reco=energy_tab[0], method=method)
+    assert_quantity_allclose(res[0, 0], 1 * data_unit)
+    assert_quantity_allclose(res[0, 1], 0 * data_unit)
+    assert_quantity_allclose(res[1, 0], 0 * data_unit)
+    assert_quantity_allclose(res[1, 1], 0 * data_unit)
+    # For the energy.nodes[20], only the the coordinates (0,0) of the (off,phi) array contains 1
+    res = bkg_3d.evaluate(fov_offset=offset_tab, fov_phi=phi_tab, energy_reco=energy_tab[1], method=method)
+    assert_quantity_allclose(res[0, 0], 0 * data_unit)
+    assert_quantity_allclose(res[0, 1], 0 * data_unit)
+    assert_quantity_allclose(res[1, 0], 0 * data_unit)
+    assert_quantity_allclose(res[1, 1], 1 * data_unit)
+
+
+def test_background2d_integrate_on_energy_band():
+    """
+    We define an energy range on witch we want to compute the integral background rate that has the same boundaries as the
+    energy axis of the Background2D one. The number of bins used for the integration on this range is the same than the
+    energy axis of the Background2D. This way, the interpolator will evaluate at energies that are the same as the one
+    that define the RegularGridInterpolator. With the method="nearest",we are sure to get 1 for the energybin where
+    are located the "1" value of the dummy Background2D data (see function make_test_cube() above). I check if for the
+    (offset) of these "1" value (23) and (59), we get what we expect by multiplying 1 by the energy bin width
+    (bin 2 for (offset)=(23) and bin 78 for (offset)=(59)).
+    """
+    bkg_2d = make_test_array()
+    energy_range = Energy([0.1, 100], 'TeV')
+    bins = 100
+    method = 'nearest'
+    off = bkg_2d.data.axis('offset').nodes
+    bkg_integrate = bkg_2d.integrate_on_energy_range(tab_energy_band=energy_range, energy_bins=bins, fov_offset=off,
+                                                     method=method)
+    data_unit = u.Unit('s-1 MeV-1 sr-1')
+    axis_energy_band = (bkg_2d.data.axes[0].bins[1:] - bkg_2d.data.axes[0].bins[:-1]).to('MeV')
+    assert_quantity_allclose(bkg_integrate[23],
+                             1 * data_unit * axis_energy_band[2])
+    assert_quantity_allclose(bkg_integrate[59],
+                             1 * data_unit * axis_energy_band[78])
+    # at 2D in spatial
+    offset_tab = Angle([bkg_2d.data.axis("offset").nodes[23].value, bkg_2d.data.axis("offset").nodes[59].value], "deg")
+    energy_tab = u.Quantity([bkg_2d.data.axis("energy").nodes.value[2], bkg_2d.data.axis("energy").nodes.value[78]],
+                            bkg_2d.data.axis("energy").nodes.unit)
+    offset_2d = np.meshgrid(offset_tab, energy_tab)[0]
+    res = bkg_2d.integrate_on_energy_range(fov_offset=offset_2d, tab_energy_band=energy_range, energy_bins=bins,
+                                           method=method)
+    assert_quantity_allclose(res[:, 0], 1 * data_unit * axis_energy_band[2])
```

Comment:
I think asserting on array shape of results is important.
The assert calls do broadcasting, so from the code it isn't clear / established what dimension / shape the returned arrays have.